### PR TITLE
Add support for Japan

### LINKED ISSUE #6479: Add support for Japan
**As a** merchant from Japan
**I want to** be able to use WCPayments
**So that** I can sell online via my Woo store.

The following is a checklist to help with this:

- [ ] Add the country to supported_countries function
- [ ] Add the country to countryFeeStripeDocsSectionNumbers function
- [ ] Onboard on Stripe using a test account form the country (make sure to share it in PR for testers)
- [ ] Make a payment using a test card
- [ ] Make sure the currency is showing properly
- [ ] Turn on multiple payment methods and make a test payment with one
- [ ] Simulate a dispute using one of the test cards. Make sure the disputes are working properly (ie, showing in the list + currency is good)
- [ ] Make sure the deposits are working properly (ie, double check with Stripe Docs the minimum deposits amount, frequency, wait time...)

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #6479

#### Changes proposed in this Pull Request
Add support for Japan
PR `3565` is required on the server side to test the fees.

#### Testing instructions
**Testing Base fees**

- Go to `WooCommerce` -> `Settings` -> `General`, and set the store Country to **Japan** and currency to **JPY**
- Place an order and verify the base fees are applied.
- Please pick a Japanese card from [Stripe's testing cards](https://stripe.com/docs/testing?testing-method=card-numbers#international-cards)

Screenshot 2023-06-29 at 19 27 23

**Testing International and FX fees**
- Go to `WooCommerce` -> `Settings` -> `General`, and set the store currency to USD
- Place an order and verify the fees are applied.
- 
Screenshot 2023-06-29 at 19 28 40

*

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6609` or branch name `add/6479-japan-fees` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/6479-japan-fees) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 003f1dabb259558c976de44b24fc1e4043123e21
- Build time: 2023-08-04 15:59:25 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/utils/account-fees.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ const countryFeeStripeDocsBaseLink =
 const countryFeeStripeDocsBaseLinkNoCountry =
 	'https://woocommerce.com/document/payments/faq/fees';
 const countryFeeStripeDocsSectionNumbers: Record< string, string > = {
+	AE: 'arabemirates',
```
Comment (anu-rock) [inline client/utils/account-fees.tsx]: IMO, UAE-related changes should not be part of this PR.

**Review by anu-rock — APPROVED** [review]
LGTM 👍 (minor comment about scope)

File: client/utils/account-fees.tsx (hunk 1 above)
Comment (mordeth) [reply client/utils/account-fees.tsx]: Sorry, it somehow landed as a merge conflict. I have removed it.
